pipeline_simulator.cpp

```diff
@@ -15,27 +15,56 @@ PipelineSimulator::~PipelineSimulator()
     // cleanup
 }
 
-void PipelineSimulator::runSimulation(const std::string &file_name, long start_inst, long inst_count, int pipelineWidth)
+void PipelineSimulator::runSimulation(const std::string &file_name, unsigned long start_inst, unsigned long inst_count, int pipeline_width)
 {
     FileInput file;
 
     file.ReadTraceFromFile(file_name);
     std::set<unsigned long> completed_instructions;
-    long executed = 0;
+
+    unsigned long executed = 0;
     while (executed < start_inst) {
         Trace tmp = file.GetNext();
         completed_instructions.insert(tmp.instructionAddr);
+
         executed++;
     }
-    executed = 0;
-    printf("%ld\n", completed_instructions.size());
-    // while (executed < inst_count) {
-    //     Trace next_instr = file.GetNext();
-    //     if (!next_instr.isValid()) {
-    //         return;
-    //     }
-    //     completed_instructions.erase(next_instr.instructionAddr);
-    //     // do something with the instruction
-    //     executed++;
-    // }
+
+    // TODO: probably need a set for when something is done ex and for mem
+    // "For dependence on integer or FP instructions, dependences are satisfied when they complete EX"
+    //"For dependence on load or store instructions, dependences are satisfied when they complete MEM"
+
+    IFStage IF_stage;
+    IDStage ID_stage;
+    EXStage EX_stage;
+    MEMStage MEM_stage;
+    WBStage WB_stage;
+    while (this->currentCycle < inst_count) {
```

Comment:
I shouldnt be checking with this->currentCycle, it runs depending on instructions executed, not cycle count.